lib/avalon/audit_objects.rb

```diff
@@ -0,0 +1,50 @@
+module Avalon
+  module AuditObjects
+
+    def self.run
+      bad_objects = []
+      ActiveFedora::Base.all.each do |object|
+        bad_objects << object unless object.valid?
+      end
+      return unless bad_objects.any?
+      AuditObjectsMailer.error_report(bad_objects, error_stats(bad_objects)).deliver_now
+    end
+
+    def self.error_stats(bad_objects)
+      # Each entry has keys :heading and :value to render as a table
+
+      # Total errors
+      stats = [ { heading: I18n.t('audit.total_header_html').html_safe,
+                  value: bad_objects.count } ]
+      object_class_counts = {}
+      collection_counts = {}
+
+      bad_objects.each do |object|
+        # Count occurences for each object class
+        object_class_counts[object.class] ||= 0
+        object_class_counts[object.class] += 1
+
+        # Count occurences for each object class
+        if object.respond_to?(:collection)
+          collection_counts[object.collection.name] ||= 0
+          collection_counts[object.collection.name] += 1
+        end
+      end
```

Comment:
Maybe rather than doing this manually, leverage built-in functionality:

```ruby
object_class_counts  = bad_objects.group_by(&:class).transform_values! {|v| v.length}
collection_counts = bad_objects.group_by {|obj| obj.collection.name }.transform_values! {|v| v.length }
```

Reply:
Nice! Needed to modify the second one to make sure `obj` responds to `collection`, but otherwise these more succinct versions drop in place.